contrib/configure-opt-thunderx2

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+#
+# Copyright (C) Mellanox Technologies Ltd. 2001-2014.  ALL RIGHTS RESERVED.
+# Copyright (C) ARM Ltd. 2016-2018.  ALL RIGHTS RESERVED.
+#
+# See file LICENSE for terms.
+#
+
+#
+# UCX build for maximal performance using specific CPU.
+# No extra debugging or profiling code.
+#
+
+basedir=$(cd $(dirname $0) && pwd)
+$basedir/configure-release \
+	CFLAGS="-O3 -mcpu=thunderx2t99 -march=armv8.1-a+lse" \
```

Comment:
looks hacky, does not allow specifying extra CFLAGS, and what about CXXFLAGS?
i would do it by autoconf, or add support for something like OPTFLAGS env var
BTW we are already using -O3 by default

Reply:
As for now I would prefer to keep it in a separate contrib script instead of autoconf. 

I think newer version of the compiler will detect the uarch and will use "-mcpu=thunderx2t99 -march=armv8.1-a+lse" out of the box. Is this correct assumption @flang-cavium  ?

You are right about CFLAGS overwrite, I will come up with something nicer.

Reply:
@flang-cavium ouch....
Sounds like we need uarch detection similar to https://gitlab.cosma.dur.ac.uk/swift/swiftsim/blob/7181b72aa7b4c38c6347ec47573029f31252e846/m4/ax_gcc_archflag.m4

Please let me know if you have better example for tx2

Reply:
```
#!/bin/bash
#
# arch-cpu-test-aarch64.sh
#

export cpuimpl=`grep 'CPU implementer' /proc/cpuinfo 2> /dev/null | cut -d: -f2 | tr -d " " | head -n 1`
export cpuarch=`grep 'CPU architecture' /proc/cpuinfo 2> /dev/null | cut -d: -f2 | tr -d " " | head -n 1`
export cpuvar=`grep 'CPU variant' /proc/cpuinfo 2> /dev/null | cut -d: -f2 | tr -d " " | head -n 1`
export cpupart=`grep 'CPU part' /proc/cpuinfo 2> /dev/null | cut -d: -f2 | tr -d " " | head -n 1`

export ax_gcc_cpu=""
export ax_gcc_arch=""

echo "cpuimpl: ${cpuimpl}"
echo "cpuarch: ${cpuarch}"
echo "cpuvar: ${cpuvar}"
echo "cpupart: ${cpupart}"

case $cpuimpl in
  0x42) case $cpupart in
    0x516 | 0x0516)
      ax_gcc_cpu="thunderx2t99"
      ax_gcc_arch="armv8.1-a+lse" ;;
    0xaf | 0x0af)
      ax_gcc_cpu="thunderx2t99"
      ax_gcc_arch="armv8.1-a+lse" ;;
    esac
    ;;

  0x43) case $cpupart in
    0x516 | 0x0516)
      ax_gcc_cpu="thunderx2t99"
      ax_gcc_arch="armv8.1-a+lse" ;;
    0xaf | 0x0af)
      ax_gcc_cpu="thunderx2t99"
      ax_gcc_arch="armv8.1-a+lse" ;;
    0xa1 | 0x0a1)
      ax_gcc_cpu="thunderxt88" ;;
    esac
    ;;
  esac

echo "ax_gcc_cpu: ${ax_gcc_cpu}"
echo "ax_gcc_arch: ${ax_gcc_arch}"

exit 0
```
On T99:

```
%> ./arch-cpu-test-aarch64.sh  
cpuimpl: 0x43
cpuarch: 8
cpuvar: 0x1
cpupart: 0x0af
ax_gcc_cpu: thunderx2t99
ax_gcc_arch: armv8.1-a+lse
```

on T88:

```
%> ./arch-cpu-test-aarch64.sh  
cpuimpl: 0x43
cpuarch: 8
cpuvar: 0x1
cpupart: 0x0a1
ax_gcc_cpu: thunderxt88
ax_gcc_arch: 
```

The 'vulcan' CPU no longer exists. That was the Broadcom name. There are no 'vulcan' CPU's out in the wild.

Reply:
@flang-cavium Okay. I would prefer to embed something like this in autoconf under --with-arm-tuned. The cpuid bits can be used for other uarch.

Reply:
> @flang-cavium Okay. I would prefer to embed something like this in autoconf under --with-arm-tuned.

Just a minor suggestion: maybe --with-aarch64-tuned instead of --with-arm-tuned.

'ARM|arm' is usually the identifier for 32-bit ARM, while 'aarch64' is the identifier for 64-bit ARM.